crates/models/codegen/src/lib.rs

```diff
@@ -0,0 +1,427 @@
+use std::{error::Error, path::Path};
+
+use ggml::Tensor;
+use llm_base::{
+    ggml, model::common, util, BasicWriteError, EvaluateOutputRequest, FileType,
+    InferenceParameters, InferenceSession, InferenceSessionParameters,
+    InferenceWithPromptParameters, KnownModel, LoadError, LoadProgress, Mmap, ModelParameters,
+    TensorLoader, TokenId, Vocabulary,
+};
+
+/// This implements [Send] and [Sync] as it is immutable after construction.
+pub struct CodeGen {
+    hyperparameters: Hyperparameters,
+    n_context_tokens: usize,
+
+    vocabulary: Vocabulary,
+
+    // normalization
+    ln_f_g: Tensor,
+    ln_f_b: Tensor,
+
+    // position embedding
+    wte: Tensor,
+
+    // language model head & bias
+    lmh_g: Tensor,
+    lmh_b: Tensor,
+
+    layers: Vec<Layer>,
+
+    inference_params: InferenceParameters,
+    inference_prompt_params: InferenceWithPromptParameters,
+
+    /// Needs to kept alive while the model is alive
+    _mmap: Option<Mmap>,
+
+    // Must be kept alive for the model
+    _context: ggml::Context,
+}
+
+unsafe impl Send for CodeGen {}
+unsafe impl Sync for CodeGen {}
+
+impl CodeGen {
+    /// Load the model from `path` with `n_context_tokens` context tokens.
+    ///
+    /// The status of the loading process will be reported through `load_progress_callback`.
```

Comment:
This is a copy-and-paste error that has been fixed in a branch - please refer to that branch for the documentation format.  https://github.com/danforbes/llama-rs/blob/dfo/chore/models/crates/models/bloom/src/lib.rs#L44-L46